src/main/java/org/mule/extension/vectors/internal/connection/provider/store/pgvector/PGVectorStoreConnection.java

```diff
@@ -0,0 +1,108 @@
+package org.mule.extension.vectors.internal.connection.provider.store.pgvector;
+
+import org.mule.extension.vectors.internal.connection.provider.store.BaseStoreConnection;
+import org.mule.extension.vectors.internal.connection.provider.store.BaseStoreConnectionParameters;
+import org.mule.extension.vectors.internal.constant.Constants;
+import org.mule.extension.vectors.internal.error.MuleVectorsErrorType;
+import org.mule.extension.vectors.internal.helper.validation.ConnectionValidationStrategies;
+import org.mule.extension.vectors.internal.util.FipsUtils;
+import org.mule.runtime.extension.api.exception.ModuleException;
+
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+import dev.langchain4j.internal.ValidationUtils;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PGVectorStoreConnection implements BaseStoreConnection {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PGVectorStoreConnection.class);
+
+  private String host;
+  private int port;
+  private String database;
+  private String user;
+  private String password;
+  private DataSource dataSource;
+  private final PGVectorStoreConnectionParameters parameters;
+
+  public PGVectorStoreConnection(PGVectorStoreConnectionParameters parameters) {
+    if (FipsUtils.isFipsEnabled()) {
+      LOGGER.warn("This vector store is not supported in GovCloud");
+    }
```

Comment:
```suggestion
    if (FipsUtils.isFipsEnabled()) {
      throw new VectorsException("This connection type is insecure and cannot be used with the fips140-2 security model.", MuleVectorsErrorType.SECURITY);
    }
```

And please create the missing VectorsException class, and the ErrorType SECURITY